src/json-uri.cpp

```diff
@@ -114,7 +114,7 @@ std::string json_uri::to_string() const
 {
 	std::stringstream s;
 
-	s << location() << " # " << pointer_.to_string();
+	s << location() << "#" << pointer_.to_string();
```

Comment:
I explicitly used spaces here, the to_string()-function should only be used for debugging (and unit-testing). At one point I had it even removed.

Or do you see a real usage for this method or the json-uri-class? 

The `json_uri`-class is really very basic and should be used extensively. Except for getting the components in the loader-callback.

Maybe it should go completely private and the loader-callback receives just a simple structure containing the fields... I don't know.

Reply:
Ok, I wondered if that was the intent. I am still a bit unclear whether the schema loader callback will only ever get URIs which don't have a fragment, and your code will take care of multiple json-pointer references into those schemas? If so, it would clarify things if the argument to the loader callback couldn't have a fragment identifier.

Reply:
Yes, schema-references should work as you say: during schema insertion a sub-schema can be referenced by different URIs. That was _the_ reason for making a new version, because impossible to resolve with 1.0.0.

Reply:
Hi @pboettch,

Sorry, that wasn't my point.

If the schema validator encounters a reference to "http://example.com/schema.json#/definitions/A", should the URL passed to the loader be just "http://example.com/schema.json" or should it have the JSON pointer?

If the schema validator subsequently encounters a reference to "http://example.com/schema.json#/definitions/B", will the loader be called again, or will the JSON pointer "/definitions/B" already have been resolved after the previous load?

Thanks!

Reply:
Sorry for my misunderstanding. The loader will only be called once per file.